src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -440,6 +425,66 @@ def _make_libraries_entry(self, target, resource_target_map, runtime_classpath):
                 library_entry["sources"] = jarred_sources.name
         return library_entry
 
+    @staticmethod
+    def _is_strict_deps(target: Target) -> bool:
+        return isinstance(
+            target, JvmTarget
+        ) and DependencyContext.global_instance().defaulted_property(target, "strict_deps")
+
+    def _flat_non_modulizable_deps_for_modulizable_targets(
+        self, modulizable_targets: FrozenOrderedSet[Target]
+    ) -> Dict[Target, FrozenOrderedSet[Target]]:
+        """Collect flat dependencies for targets that will end up in libraries. When visiting a
+        target, we don't expand the dependencies that are modulizable targets, since we need to
+        reflect those relationships in a separate way later on.
+
+        E.g. if A -> B -> C -> D and A -> E and B -> F, if modulizable_targets = {A, B}, the resulting map will be:
+         {
+            A -> {E},
+            B -> {C, F, D},
+
+            // Some other entries for intermediate dependencies
+            C -> {D},
+            E -> {},
+            F -> {},
+         }
+        Therefore, when computing the library entries for A, we need to walk the (transitive) modulizable dependency graph,
+        and accumulate the entries in the map.
+
+        This function takes strict_deps into account when generating the graph.
+        """
+        flat_deps = {}
+
+        def create_entry_for_target(target: Target) -> None:
+            target_key = target
+            non_modulizable_deps = [
+                dep for dep in target.dependencies if dep not in modulizable_targets
```

Comment:
Should this be using `strict_dependencies` instead of `dependencies` if `self._is_strict_deps(dep)`?

Reply:
Good point! In my mental model those two things are equivalent, but I don't have a full picture of strict_deps yet.